packages/varlet-ui/src/uploader/Uploader.vue

```diff
@@ -181,15 +181,20 @@ export default defineComponent({
     )
 
     function preview(varFile: VarFile) {
-      const { disabled, previewed } = props
+      const { disabled, previewed, preventDefaultPreview, onPreview } = props
+
+      if (preventDefaultPreview) {
+        call(onPreview, reactive(varFile))
+        return
+      }
 
       if (form?.disabled.value || disabled || !previewed) {
         return
       }
 
       const { url } = varFile
 
-      if (isHTMLSupportImage(url)) {
+      if (url && isHTMLSupportImage(url)) {
```

Comment:
don't need to check the url, isHTMLSupportImage already does that.

Reply:
ok~